HLint.hs

```diff
@@ -11,3 +11,4 @@ ignore "Use <$>"
 ignore "Redundant bracket"
 ignore "Use lambda-case"
 ignore "Use $>" -- added temporarily
+ignore "Evaluate"
```

Comment:
There are a lot of ignored suggestions now. I wonder if we actually want all of them to be applied uniformly across the code base or if some of these could be applied only to specified modules (`{-# ANN module "HLint: ignore Evaluate" #-}`).

Reply:
Never really liked Evaluate. In the specific case it's a disaster:

```
./src/Pact/Compile.hs:268:16: Warning: Evaluate
Found:
  TTable (TableName tn) (fst cm) (snd cm)
Why not:
  uncurry (TTable (TableName tn)) cm
```
The code is clearly ill-served by using `uncurry`:
```
      return $ TTable (TableName tn) (fst cm) (snd cm) tty docs i
```

But in general I'm not fond of the suggestion unless your code is particularly pair-heavy with Arrow all over it, etc.

Reply:
I agree overall that we might want to discuss the ignored suggestions -- there are some I find very useful (like unnecessary parens) that we have ignored.